### ISSUE TITLE
Mixture of validation and evaluation 

### ISSUE THREAD

**Issue by MinWang1997** [pytorch/ignite#2304]
## ❓ Questions/Help/Support

Hi,

I found that you use test set as validation and evaluation
E.g., in examples/contrib/cifar10/main.py

Usually during the training process, the model should not touch the test data and should use validation set for validation and then we evaluate the model on test dataset.

Best wishes,
Min
